test/e2e_flags_test.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"flag"
+	"testing"
+)
+
+func TestE2eFlags(t *testing.T) {
+	v := flag.Lookup("v")
+	if v == nil {
+		t.Fatal("Could not find 'v' flag; klog was not init-ed")
+	}
+	if v.Value.String() != klogDefaultLogLevel {
+		t.Fatalf("Either '%s' was passed in for v or v is not being initialized properly. The former is a test limitation, but the latter is an actual problem", v.Value.String())
+	}
+	if flagsSetup {
+		t.Error("SetupLoggingFlags should not have been called by the unit tests")
+	}
+
+	var flagTests = []struct {
+		testCase    string
+		logVerbose  bool
+		nonDefaultV string
+		expectedV   string
+	}{
+		{
+			testCase:    "Check default case with logverbose off",
+			logVerbose:  false,
+			nonDefaultV: "",
+			expectedV:   klogDefaultLogLevel,
+		},
+		{
+			testCase:    "Check default case with logverbose on",
+			logVerbose:  true,
+			nonDefaultV: "",
+			expectedV:   "8",
+		},
+		{
+			testCase:    "Check default case with logverbose on and v set to non-default value",
+			logVerbose:  true,
+			nonDefaultV: "5",
+			expectedV:   "5",
+		},
+	}
+
+	alsologtostderr := flag.Lookup("alsologtostderr").Value.String()
+	if "true" != alsologtostderr {
+		t.Errorf("alsologtostderr was '%s', expected 'true'\n", alsologtostderr)
+	}
+
+	for _, tt := range flagTests {
```

Comment:
Rather than logging `tt.testCase`, the standard way is to do
```
for _, tc := range tests {
  t.Run(tc.testCase, func(t *testing.T) {
    ...
  })
}
```

Reply:
done, thanks.